keps/77-dynamically-sized-jobs/README.md

```diff
@@ -7,168 +7,283 @@
+In the context of WorkloadSlice scheduling, when a workload qualifies for multiple resource flavors, i.e., more than one flavor can satisfy the resource requirements and selector constraints, the scheduler must determine which flavor to assign. 
+Kueue currently does not support combining resources from multiple flavors within a single PodSet; however, multiple PodSets within the same workload may be admitted using different resource flavors.
 
-### Garbage Collecting Workload Slices
+Flavor selection is influenced by factors such as quota availability, taint tolerations, node label selectors, and, where applicable, topology-aware scheduling policies.
 
-The logic of folding and deleting would be isolated in this controller. We don’t necessarily have to say that this Workload Slice belongs to a specific pod. This controller will look at the Workload objects and check whether they have the Admitted condition or not.
+When scaling up a previously admitted workload slice, the current implementation enforces a sticky flavor constraint, meaning that the new slice must reuse the originally assigned flavor, even if other eligible flavors have available capacity. 
+This approach avoids unpredictable flavor switching but may lead to the new slice remaining in a Pending state if the original flavor lacks sufficient resources.
 
-1. The controller increments the `.status.admission.podSetAssignments.count` and passes the UID of the workload you are folding to the parent workload.
-If we still don’t see the workload being deleted, we at least know it has been counted towards the parent workload and the controller shouldn't fold it again.
+Future enhancements may introduce support for controlled flavor reassignment or user-defined policies to express preferences or tolerances for such transitions.
 
-## Phases for MVP (alpha)
+#### Garbage Collection of Preempted Workload Slices
+Currently, a Workload can become outdated, marked by Kueue as "out-of-sync", and subsequently transitioned to the "Finished" state.
+This also applies to outdated Workload slices: when multiple successive updates are issued for a given Job, any superseded slices are marked as "Finished" to reflect their obsolescence.
 
-### Phase 1 - Scale Down
-Scaling down will be the first phase towards MVP because it can be implemented without introducing the Workload Slices. 
+Preempted (i.e., aggregated) Workload slices are marked as Finished.
+Under the current design proposal, all finished workload slices are retained indefinitely and are not garbage collected.
+To address potential resource buildup, a `PreemptedWorkloadSliceHistory` mechanism whether as a configuration option or an API field on the Workload—could be introduced to limit the number of retained inactive slices, 
+similar to how `revisionHistoryLimit` is used in Kubernetes Deployments to manage ReplicaSet history.
```

Comment:
Let's defer this to Beta (add the graduation point). For alpha, if we aim 0.13 we will not have time for it. Also, I think ObjectRetentionPolicy should handle it in most cases.

Reply:
Agreed, it is already in "Beta" section.

Reply:
For posterity, as [currently defined](https://kueue.sigs.k8s.io/docs/tasks/manage/setup_object_retention_policy/#set-up-a-retention-policy):

```yaml
objectRetentionPolicies:
  workloads:
    afterFinished: "1h"
    afterDeactivatedByKueue: "1h"
```

the `objectRetentionPolicy` is exclusively duration-driven, which may not be the best fit for garbage collecting *aggregated* workload slices.

As you suggested, adding a `finishedWorkloadLimit` (or similar) field to the policy definition would be preferable.